firmware/quadruna/VC/src/app/vehicle_dynamics/app_regen.c

```diff
@@ -119,51 +161,26 @@ void app_regen_computeActiveDifferentialTorque(ActiveDifferential_Inputs *inputs
     regenAttr->right_inverter_torque_Nm = torque_right_Nm * scale;
 }
 
-static void computeRegenTorqueRequest(
-    ActiveDifferential_Inputs *activeDiffInputs,
-    RegenBraking_Inputs       *regenAttr,
-    PowerLimiting_Inputs      *powerInputs)
+void app_regen_sendTorqueRequest(float left, float right)
 {
-    float pedal_percentage = activeDiffInputs->accelerator_pedal_percentage;
-    float min_motor_speed =
-        MOTOR_RPM_TO_KMH(MIN(activeDiffInputs->motor_speed_right_rpm, activeDiffInputs->motor_speed_left_rpm));
-
-    powerInputs->accelerator_pedal_percent = -pedal_percentage; // power limiting function requires positive pedal value
-    powerInputs->left_motor_temp_C         = app_canRx_INVL_MotorTemperature_get();
-    powerInputs->right_motor_temp_C        = app_canRx_INVR_MotorTemperature_get();
-
-    regenAttr->derating_value = 1.0f;
+    app_canTx_VC_LeftInverterTorqueCommand_set(left);
+    app_canTx_VC_RightInverterTorqueCommand_set(right);
+}
 
-    if (regenAttr->battery_level > 3.9f)
-    {
-        regenAttr->derating_value = SOC_LIMIT_DERATING_VALUE;
-    }
+float app_regen_pedalRemapping(float apps_pedal_percentage)
+{
+    apps_pedal_percentage = (apps_pedal_percentage - PEDAL_SCALE);
 
-    if (min_motor_speed <= 10.0f)
+    if (apps_pedal_percentage < 0.0f)
     {
-        regenAttr->derating_value = (min_motor_speed - SPEED_MIN_kph) / (SPEED_MIN_kph);
+        return apps_pedal_percentage / PEDAL_SCALE;
     }
-
-    activeDiffInputs->power_max_kW    = app_powerLimiting_computeMaxPower(powerInputs);
-    activeDiffInputs->wheel_angle_deg = app_canRx_FSM_SteeringAngle_get() * APPROX_STEERING_TO_WHEEL_ANGLE;
-
-    if (regenAttr->enable_active_differential)
+    else if (apps_pedal_percentage < 0.1f)
     {
-        app_regen_computeActiveDifferentialTorque(activeDiffInputs, regenAttr);
+        return 0.0f;
     }
     else
     {
-        // no power limit, no active differential
-        regenAttr->left_inverter_torque_Nm  = MAX_REGEN_Nm * pedal_percentage * regenAttr->derating_value;
-        regenAttr->right_inverter_torque_Nm = MAX_REGEN_Nm * pedal_percentage * regenAttr->derating_value;
+        return apps_pedal_percentage / (MAX_PEDAL_PERCENT - PEDAL_SCALE - 0.1f);
```

Comment:
i don't think this is right, I think you need to
```suggestion
        return (apps_pedal_percentage-0.1f) / (MAX_PEDAL_PERCENT - PEDAL_SCALE - 0.1f);
```

Reply:
because this assumes that the increasing part of the graph is rooted at the origin

Reply:
fak i should write tests for this